s3po/backends/memory.py

```diff
@@ -14,6 +14,13 @@ def __init__(self):
     def download(self, bucket, key, fobj, retries, headers=None):
         '''Download the contents of bucket/key to fobj'''
         obj = self.buckets[bucket].get(key)
+
+        # if we stored a bytes object, decode it into a string
+        try:
+            obj = obj.decode('ascii')
```

Comment:
I think instead of this you should change the mode for `download_file` to `wb` so that the file object will be in binary mode as well.

Reply:
I wish that was the fix. The test failure that needed fixing was on https://github.com/seomoz/s3po/blob/master/test/test_connection.py#L29 where it checks the result of `self.conn.download`. For the `memory` backend, `download()` grabs the actual python object that is stored, so when `upload_file` runs with mode `rb`, the value that the memory backend stores is `bytes`.

https://github.com/seomoz/s3po/blob/master/s3po/backends/memory.py#L20 is trying to write those bytes to into `fobj` which is a `StringIO` from https://github.com/seomoz/s3po/blob/master/s3po/connection.py#L79.

I'm totally not sure what the "right way" to handle the memory backend is in the face of string vs. bytes.

Reply:
It looks like the default for `StringIO` is to use the system locale via this method:
```
locale.getpreferredencoding(False)
```
I think that's preferable to specifying `ascii` since that could throw away bytes greater than 0x7f.